test/MUIDataTableBody.test.js

```diff
@@ -271,6 +271,42 @@ describe('<TableBody />', function() {
     assert.strictEqual(toggleExpandRow.callCount, 0);
   });
 
+  it('should not gather selected row data when clicking row with selectableRowsOnClick=true when it is disabled with isRowSelectable via selectedRows.', () => {
```

Comment:
Thank you for adding this! Additional tests are always appreciated.

To keep the types in line with what's used in the table, how about changing the `selectedRows` argument to contain `{ lookup: { dataindex: 2 }, data: [{ dataindex: 2, index: 2 }] }` and then updating `selectedRows.length` in the `isRowSelectable` function to `selectedRows.data.length`?